build_tools/cmake/iree_copts.cmake

```diff
@@ -427,7 +427,20 @@ endif()
 # Compiler: MSVC
 #-------------------------------------------------------------------------------
 
-# TODO(benvanik): MSVC options.
+if(MSVC)
+  if("${CMAKE_C_COMPILER_LAUNCHER}" MATCHES "ccache" OR
+     "${CMAKE_CXX_COMPILER_LAUNCHER}" MATCHES "ccache")
+    # Disable separate PDB file generation (for debug info) when using ccache.
+    # ccache silently falls back to the real compiler when an unsupported flag
+    # like /Zi is encountered.
+    message(STATUS "Replacing /Zi with /Z7 since ccache is in use and does not support /Zi")
+    # https://learn.microsoft.com/en-us/cpp/build/reference/z7-zi-zi-debug-information-format
+    string(REPLACE "/Zi" "/Z7" CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG}")
+    string(REPLACE "/Zi" "/Z7" CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG}")
+    string(REPLACE "/Zi" "/Z7" CMAKE_C_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS_RELWITHDEBINFO}")
+    string(REPLACE "/Zi" "/Z7" CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO}")
```

Comment:
Note: this will affect downstream users too. Not really a fan of when projects modify global flags, but I think this is niche enough...

Reply:
since it's conditional on ccache usage and anyone using ccache would need the same thing it seems ok - may still conflict with a user's similar hacks but probably worth waiting for that to be an issue :)